js/storage.ts

```diff
@@ -0,0 +1,33 @@
+export interface Storage {
+  readonly length: number;
+
+  key(index: number): string;
+  getItem(key: string): string;
+  setItem(key: string, value: string): void;
+  removeItem(key: string): void;
+  clear(): void;
+}
+
+export class SessionStorage implements Storage {
+  private _storage: Record<string, string> = {};
+
+  get length(): number {
+    return Object.keys(this._storage).length;
+  }
+
+  key(index: number): string {
+    return Object.keys(this._storage)[index];
+  }
+  getItem(key: string): string {
```

Comment:
Would it be revelant to return an empty string instead of an undefined value in this case?

Reply:
@zekth I don't think so. Maybe user put `""` there? `undefined` will let user see that it's... undefined!

Reply:
Both solution are good, it's more like a philosophy. Some APIs returns empty strings, others undefined as the return type is a string. Was just wondering.